[t1212] - Auto load saved data

secretrobotron opened the pull request:
1. Removed majority of templates/test.js
2. Added XHR for saved data and default config option to specify where it is
3. Added saved-data.json

secretrobotron committed:
[t1212] - Auto load saved data

1. Removed majority of templates/test.js
2. Added XHR for saved data and default config option to specify where it is
3. Added saved-data.json

jbuck commented on src/main.js:
```diff
@@ -661,6 +661,38 @@
       }
 
+      function attemptDataLoad( finishedCallback ){
+        var xhr = new XMLHttpRequest(),
```
Why not use my XHR module?

secretrobotron replied:
Because it wasn't as robust as the code that Dave added for loading configs, which was incredibly simple to copy & paste from right below this code. Perhaps we can merge that code in for a better XHR wrapper.

secretrobotron committed:
[t1212] tests for saved data auto-loading

1. Added tests for auto-loading
2. Added config option to skip page scrape
3. Fixed broken test

mjschranz commented on test/core.js:
```diff
@@ -857,4 +859,19 @@
+    Butter({
+      config: "test-config-auto-load.json",
+      ready: function( butter ){
+        start();
```
this call should be after all the tests right?

secretrobotron replied:
Actually, I'm not sure this matters. I checked some documentation, and nowhere does it say `start()` must be in a specific place.

Also (from start function in qunit.js):

``` javascript
        // A slight delay, to avoid any current callbacks
        if ( defined.setTimeout ) {
            window.setTimeout(function() {
                if ( config.timeout ) {
                    clearTimeout(config.timeout);
                }

                config.blocking = false;
                process();
            }, 13);
        } else {
            config.blocking = false;
            process();
        }
```

secretrobotron replied:
To make it easy on the eyes though, I'll certainly change it :).

secretrobotron committed:
[t1212] review change - moved start() in test

secretrobotron committed:
[t1212] review changes

1. Removed saved-data.json for saveDataUrl in default config
2. Provided method to avoid saveDataUrl functionality entirely in main.js

dseif commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/ead87e387260871/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/ead87e387260871/output.txt

Total script time: 0.58 mins
